src/main/ApplicationImpl.cpp

```diff
@@ -439,10 +439,13 @@ ApplicationImpl::validateAndLogConfig()
 
     if (getHistoryArchiveManager().hasAnyWritableHistoryArchive())
     {
-        if (!mConfig.MODE_STORES_HISTORY)
+        if (!mConfig.MODE_STORES_HISTORY ||
```

Comment:
see my other comment: what we're looking for here is to check that if publish is enabled we better store all historical data. So a helper function here would help